lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -46,7 +46,6 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
-          warn_about_bare_error if warning_about_bare_error && !negative_expectation
           return false unless Proc === given_proc
```

Comment:
The user is properly notified of their mistake.  See 79582c2c.  Changing to raise an ArgumentError would be a breaking change.